include/deal.II/dofs/dof_tools.h

```diff
@@ -1760,6 +1760,25 @@ namespace DoFTools
                       const bool            invert_vertex_mapping  = false);
 
   /**
+   *  Same as above but allows boundary dofs on blocks to be excluded individually
+   *
+   * @arg <tt>exclude_boundary_dofs</tt>:  for each patch of cells around a
+   * vertex, collect all of the interior degrees of freedom of the patch and
+   * disregard those on the boundary of the patch if the boolean value for
+   * the corresponding block in the BlockMask is false.
```

Comment:
I don't actually understand what this is doing. Can you provide an example for what you want?